packages/dumb/src/components/settings/history/history-table.util.ts

```diff
@@ -0,0 +1,130 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {
+  ChangeType,
+  HistorizedParameter,
+  IncomingRow,
+  Parameter,
+  ParameterValue
+} from '../../../models/historized-parameter.model'
+import { formatLocalizedFromUTC, getLongDayHourFormat } from '../../../utils/datetime/datetime.util'
+import { TimePrefs, Unit } from 'medical-domain'
+
+const setCurrentParameter = (parameter: Parameter, currentParameters: Map<string, ParameterValue>): void => {
+  switch (parameter.changeType) {
+    case ChangeType.Added:
+    case ChangeType.Updated:
+      currentParameters.set(parameter.name, {
+        value: parameter.value,
+        unit: parameter.unit
+      })
+      break
+    case ChangeType.Deleted:
+      if (currentParameters.has(parameter.name)) {
+        currentParameters.delete(parameter.name)
+      }
+      break
+    default:
+      break
+  }
+}
+
+const getRowByParameter = (parameter: Parameter, timePrefs: TimePrefs, currentParameters: Map<string, ParameterValue>): HistorizedParameter => {
+  const dateFormat = getLongDayHourFormat()
+  const row: Partial<HistorizedParameter> = { ...parameter }
+  row.rawData = parameter.name
+  const changeDate = new Date(parameter.effectiveDate)
+
+  row.parameterDate = formatLocalizedFromUTC(changeDate, timePrefs, dateFormat)
+
+  if (parameter.changeType === ChangeType.Updated) {
+    if (currentParameters.has(parameter.name)) {
+      const currentParam = currentParameters.get(parameter.name)
+      if (currentParam) {
+        row.previousUnit = currentParam.unit
+        row.previousValue = currentParam.value
+      }
+    } else {
+      row.changeType = ChangeType.Added
+    }
+  }
+
+  setCurrentParameter(parameter, currentParameters)
+
+  row.isGroupedParameterHeader = false
+
+  return row as HistorizedParameter
+}
+
+const getLatestDate = (parameters: Parameter[]): Date => {
+  return parameters.reduce((currentLatestDate: Date, parameter: Parameter) => {
+    const changeDate = new Date(parameter.effectiveDate)
+    return currentLatestDate.getTime() < changeDate.getTime() ? changeDate : currentLatestDate
+  }, new Date(0))
+}
+
+export const buildAllRows = (history: IncomingRow[], timePrefs: TimePrefs): HistorizedParameter[] => {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (blocking):** ‏I see high value in this function being tested (UT). Don't you agree?